### PR TITLE
Refactor test translation

### PR DESCRIPTION
**Pull request by atilaneves**

### CODE REVIEW

File: translate/tests/test_asttoirvisitor.py [hunk 1]
```diff
@@ -1,12 +1,9 @@
-#!/usr/bin/env python3
-import unittest
```
Comment (marler8997) [inline translate/tests/test_asttoirvisitor.py]: I thought you wanted this to be a python unittest?

File: translate/pytest_translate [hunk 2]
```diff
@@ -2,63 +2,37 @@
 #
 # Translate a python test to another language.
 #
-import enum
```
Comment (marler8997) [inline translate/pytest_translate]: Why did you remove the language differentiation logic?  Isn't the intention to support multiple languages?  `pytest_translate` is supposed to be able to translate multiple languages, not just `C#`.

**Comment by marler8997** [comment]
Are there plans for Windows support?

File: translate/pytest_translate (hunk 2 above)
Comment (atilaneves) [reply translate/pytest_translate]: Yes, but no other implementations exist yet.

**Comment by atilaneves** [comment]
> Are there plans for Windows support?

We should run the tests on Windows as well, but that's a CI scripting issue, no? The biggest problem there is that this is a symmetryinvestments repo, and I can't sign up for AppVeyor for that.

File: translate/tests/test_asttoirvisitor.py (hunk 1 above)
Comment (atilaneves) [reply translate/tests/test_asttoirvisitor.py]: pytest

**Comment by Laeeth** [comment]
Can you see this?
https://ci.appveyor.com/project/Laeeth/autowrap/events

File: translate/pytest_translate (hunk 2 above)
Comment (marler8997) [reply translate/pytest_translate]: Right but the logic is there to detect the language from the file extension. If you try to translate to another language it will detect it and give you a nice error message. Why remove that functionality?

Comment (marler8997) [reply translate/pytest_translate]: With the current implementation, if I run
```
./translate/pytest_translate tests/test_simple.py foo.java
```
I'll get this error message:
```
ERROR: unknown file extension from 'foo.java'
```

With your removal, we are no longer detecting the language, we will just generate `C#` code to put in `foo.java`.  If you don't want to use the file extension to detect the language, then we should add a command-line option, we shouldn't remove the language option because now the user will have unexpected results, like generate `C#` in a Java source file.

Comment (Laeeth) [reply translate/pytest_translate]: It's an empirical question as to whether YAGNI and in a sense also an optimisation problem in the face of uncertainty and through time where the right choice depends on the consequence of removing code that actually you are going to need versus cluttering up the code with maybes.

The dimension of those choices within a commercial environment is dollars though the expression may be a deeply technical one.  Ie quality and clutter - they are important and do have often neglected dollar consequences.

But so does deleting stuff you definitely are going to need.  The cost of the latter depends on the cost of putting the stuff that was deleted back when it turns out to be needed, just as you reckoned.

Much depends on the context too.  If requirements are specified by managers who in truth do not really know what's going to be needed, then YAGNI is a sensible default.  So it's probably a sensible default in almost every context, just not absolutely every context and there are exceptions.

Here, on the whole, I think I know better than users and practitioners on the business side what they are going to need, with an unfair advantage coming from the fact that I have quite some say over what they are going to be needing!

So it's my opinion that a different balance would be the right one in this context.  I haven't been able to really succeed in having a discussion about that though, one reason being that remote does make it harder to recognise the context and another being that negative slack makes it harder to do the necessary work in communicating the context.

Comment (marler8997) [reply translate/pytest_translate]: I won't argue that we are 100% going to support multiple languages.  However, I do think that we should detect this use case and error rather than blindly generate `C#` code in a non `.cs` file:
```
./pytest_translate test_simple.py foo.java
```

The code being removed here is only detecting the language and printing an error when an unsupported language is detected, it does not contain implementation for any other language.

**Comment by atilaneves** [comment]
> Can you see this?
> https://ci.appveyor.com/project/Laeeth/autowrap/events

No, but I also have appveyor for my fork, just not the main repo.

File: translate/pytest_translate (hunk 2 above)
Comment (atilaneves) [reply translate/pytest_translate]: It's still the only implementation that exists. At most, it should raise an exception if the file extension isn't `.cs`, and even if different backends existed, the enum wasn't necessary.

File: translate/pytest_translate [hunk 3]
```diff
@@ -2,63 +2,41 @@
...
-
-class UnknownFilenameLanguageError(Exception):
-    def __init__(self, filename):
-        self.filename = filename
 
+def main():
+    args = cmd_line_args()
+    extension = os.path.splitext(args.output_file)[1]
+    if extension != '.cs':
+        raise Exception(f"Unsupported file extension '{extension}'")
```
Comment (marler8997) [inline translate/pytest_translate]: Why raise an exception here?  The error isn't a code bug, it's a user error, so they don't need to see a stack trace.
```python
sys.exit("ERROR: unsupported output file extension '{extension}'")
```
